django/utils/http.py

```diff
@@ -296,8 +296,6 @@ def is_safe_url(url, allowed_hosts=None, require_https=False):
         url = url.strip()
     if not url:
         return False
-    if allowed_hosts is None:
```

Comment:
It might be useful to continue allow passing `None`. That feels a bit more natural to me than passing `set()`.